src/rain-report/rain-report.service.ts

```diff
@@ -1,22 +1,28 @@
 import { Injectable } from '@nestjs/common';
 import { RainReport } from '@prisma/client';
-import { PrismaClient } from '@prisma/client';
+import { DatabaseService } from './database.service';
 
 @Injectable()
 export class RainReportService {
+  constructor(private readonly database: DatabaseService) {}
+
   /**
    * Retrieve all `RainReport` records from the database.
    */
   async all(): Promise<Partial<RainReport>[]> {
-    // TODO: create a service and inject
-    const database = new PrismaClient();
-    try {
-      return await database.rainReport.findMany({
-        select: { timestamp: true, rain: true },
-        orderBy: { timestamp: 'desc' },
-      });
-    } finally {
-      await database.$disconnect();
-    }
+    return await this.database.rainReport.findMany({
+      select: { rain: true, timestamp: true },
+      orderBy: { timestamp: 'desc' },
+    });
+  }
+
+  /**
+   * Create a new `RainReport` record for the current
+   * datetime and return the created `RainReport` record.
+   */
+  async create(isRaining: boolean): Promise<RainReport> {
```

Comment:
Implementation of `RainReportService.create()`. Unlike the `list()` method this returns the newly created `RainReport` as-is and does not exclude the `id` (which was specified in the example response payload for `GET /api/data`).